src/sql/inspect.py

```diff
@@ -44,6 +44,32 @@ def __init__(self, schema=None, conn=None) -> None:
         self._table_txt = self._table.get_string()
 
 
+def _fill_nulls(keys, mapping):
+    out = dict()
+
+    for key in keys:
+        if key not in mapping:
+            out[key] = ""
+        else:
+            out[key] = mapping[key]
+
+    return out
+
+
+# we're assuming there's one row that contains all keys, I tested this and worked fine
+# my initial implementation just took all keys that appeared in "rows" but then order
+# isn't preserved, which is important for user experience
+def _get_keys(rows):
```

Comment:
wdym by typesafe?

keys and values are always strings

I don't think rows can be None or empty, because that would be a table with no columns, but I added a check and test case

Reply:
type hints don't enforce type validation, they are only helpful for linting tools. I'm ok adding them but then we should put these things in writing. are we expecting them always, under certain conditions, or are they optional?

Reply:
see note at the top: https://docs.python.org/3/library/typing.html